src/function/table/table_scan.cpp

```diff
@@ -321,9 +324,23 @@ class DuckTableScanState : public TableScanGlobalState {
 			}
 
 			auto next = storage.NextParallelScan(context, state, l_state.scan_state);
+			if (data_p.results_execution_mode == AsyncResultsExecutionMode::TASK_EXECUTOR) {
+				// We can avoid looping, and just return as appropriate
+				if (!next) {
+					data_p.async_result = AsyncResultType::FINISHED;
```

Comment:
Unrelated to this PR specifically - but I'm not sure why we have `AsyncResultType` as a separate enum from `SourceResultType`. The only difference seems to be the addition of `IMPLICIT`. Can't we unify them and add `SourceResultType::IMPLICIT` to `SourceResultType`?

I just don't really see the need to call this `async_result` or to have an `AsyncResultType` when nothing here has anything to do with `async` - it just seems confusing. imo this should just be `result_type` and `SourceResultType::HAVE_MORE_OUTPUT`.